src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_NUMBER_OF_FIGURES = 5;
+    public static final double MAX_SIZE = 100d;
+    public static final double DEFAULT_RADIUS_OF_CIRCLE = 10d;
+    public static final String DEFAULT_COLOR = "white";
+
+    private final Random random;
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier(Random random, ColorSupplier colorSupplier) {
+        this.random = random;
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(MAX_NUMBER_OF_FIGURES)) {
+            case 0:
+                double radius = random.nextDouble() * MAX_SIZE;

```

Comment:
> can we introduce a new method for `random.nextDouble() * MAX_SIZE;` ? that will let us avoid code duplicates below.

Does that make sense? Instead of this line, we will have a duplicate method name. The code will not take up less space